app/views/businesses/_form.html.erb

```diff
@@ -60,7 +60,9 @@
     </div>
 
     <div class="flex justify-end mt-8 mb-16 mr-4 sm:mr-6 lg:mr-0">
-      <%= form.submit t(".save"), class: "ml-3 inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-gray-600 hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-500" %>
+      <%= form.button class: "ml-3 inline-flex justify-center align-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-gray-600 hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-500" do %>
+        <%= render SaveButtonComponent.new(text: t(".save"), disable_with: t(".processing")) %>
+      <% end %>
```

Comment:
This diff is making me wonder if we are making the wrong abstraction here. Should the component instead be the _entire_ button, not just the stuff inside? It could take in the form to chain the helper. That would make these really nice to use because _all_ of the CSS would be in there. If you need to add `ml-3` or something very specific we could pass in an optional parameter.

Reply:
Unfortunately a big known issue with View Component right now is that it doesn't play nice with Rails' form builder. We'd have to make a custom form builder to achieve this.

Reply:
What do you mean? We can't pass in the `form` object?

Reply:
I'm not entirely sure about the specifics. I'm just familiar with some issues I've read on the repo and [this note on the package's website](https://viewcomponent.org/known_issues.html#form_for-compatibility).

Reply:
Ah, I see. https://github.com/github/view_component/discussions/420

The main issue is passing in the `form` couples the component with the form builder object. On the surface that seems fine, but if we ever need to storybook or use them outside of a form context it starts to fall apart.

I'm not sure what the best approach is, here, to be honest. Maybe it is to keep what we have and not abstract any father?

Reply:
OK, messing around with some code for #203, turns out we don't need a reference to the form at all. We can use `submit_tag` or `button_tag type: :submit` and it will submit the `<form>` that it is nested under.

It doesn't solve the whole "ViewComponent + FormBuilder" but we can punt on that for now!

Reply:
> This diff is making me wonder if we are making the wrong abstraction here. Should the component instead be the _entire_ button, not just the stuff inside? It could take in the form to chain the helper. That would make these really nice to use because _all_ of the CSS would be in there. If you need to add `ml-3` or something very specific we could pass in an optional parameter.

I had the same question while doing it, but I just went for an approach focused more on the content and not coupled with the form itself. But using the `button_tag` helper sounds good and keeps the component clean.